### ISSUE TITLE
Pressing and dragging the title bar doesn't actually drag the window

### ISSUE THREAD

**Issue by Risuleia** [Vadoola/Tomotroid#60]
#### Problematic Code:
```rust
out property <bool> title-bar-pressed: false;

// ... rest of the code

TouchArea {
     moved => {
          if (root.title-bar-pressed) {
                 root.move();
          }
    }
    pointer-event(ev) => {
           root.title-bar-pressed = (ev.kind == PointerEventKind.down) && (ev.button == PointerEventButton.left);
    }
}
``` 
> This above mentioned part of the code is in the Text component for the custom title bar in the Borderless Window component.

#### The Reason:
Although, the logic for it doesn't seem wrong overall. What must be considered, that the stated check doesn't take into account the movement of the mouse. Upon moving by as little as 1px, the pointer-event is reset, and thus the check never returns `true`.
Therefore, as per the logic of the above code, the `title-bar-pressed` property becomes `true` **if and only if** the <ins>the left mouse button is being pressed and the mouse position doesn't change</ins>, and hence, the window dragging logic is never called continuously, taking away the entire essence of "dragging".

#### Suggested Fix:
Instead of setting a custom property upon meeting specific conditions, it's better to check if the TouchArea is being inherently pressed, and if so, then to call upon the dragging logic.

```rust
TouchArea {
     moved => {
           if (self.pressed && self.enabled) {
                  root.move()
           }
      }
}
```
The `self.pressed` returns `true` whenever the TouchArea is being pressed. Since the condition is being checked under the `move` event, the point-event doesn't get reset upon any change in mouse position.
> Additionally, there's a condition `self.enabled`, just in place to make sure the titlebar is enabled.
___
### Issue encountered on:
**Operating System:** Windows 11 Pro (Version 22H2)
**Processor:** AMD Ryzen 5 5500U 2.10GHz, with integrated Radeon graphics
**RAM:** 8GB
> Although, it's probably not an issue specific to certain machines or operating systems, I attached the device information just in case.
___
Hope it helps!

**Comment by Vadoola** [comment]
Honestly I don't recall why I wrote the title bar move logic exactly the way I did, as it was some of the earliest code I wrote on this project.

However, if I'm understanding you correctly, you are saying the Windows never actually moves for you?
That certainly is strange, because while the logic may not be perfect, and probably needs some fixing (See Issue #28, and Issue #52, for two different issues relating to the title bar dragging logic, one that happens on Wayland, and one that happens on Windows), the window does drag around on every system I've tested in on, including Windows 10, 11, Linux under Wayland (and I'm pretty sure I tested under X11 at one point as well).

It's very possible your suggested logic change could be better, and is worth testing to find out, but would you be able to upload a video of what you are seeing in the current logic when it comes to dragging the window around? I want to make sure I understand correctly what you are seeing.

**Comment by Risuleia** [comment]
Sure, here's a video:

https://github.com/Vadoola/Tomotroid/assets/72485538/c01d08fb-6937-40eb-ba20-32511ef2141b

**Comment by Vadoola** [comment]
Interesting, that's certainly not the results I'm seeing. I'm not in front of my dev machine right now, so I can't take a video of it working on mine.

I wonder what could be different? I've been pretty busy the last couple of weeks, and what little time I've put in has been on getting the Global Hot Keys working, so I haven't updated to Slint 1.4.x. What version of Slint are you compiling with? I wonder if there was an update in event handling in 1.4.x that is making the difference?

Regardless it could very well be a change worth making, I never dug into Issue #28, just documented that it was happening, but I imagine the title-bar-pressed not getting updated properly on mouse release could be what is causing the problem documented in Issue #28, and your change should eliminate that.

I'll see if I can get some time to the make the modification and do some tests. Thanks for the suggestion.

P.S. The design of your app Tranquilo looks nice.

**Comment by Risuleia** [comment]
It could be I suppose. I am using Slint 1.4.1. 
Take it at your own pace!

And thanks :).

**Comment by Vadoola** [comment]
Yeah, so I'm not entirely sure what specifically changed, I haven't really looked into it, but it was something that changed between Slint 1.3.2 (what I had been using) and Slint 1.4.1. Simply upgrading to 1.4.1 broke moving the Window, and your change fixes it. Sadly it doesn't actually fix the Wayland issue was I was having as described in #28, so I'll have to dig into that more at some point.

I appreciate you pointing this out to me.


This is closed as of commit 025abd3 with the upgrade to Slint 1.4.1 and the adjusted move logic as suggest by Risuleia.
